2-AvailabilityCheck/Get-AvailabilityInformation.ps1

```diff
@@ -48,583 +48,626 @@
     - Requires Azure PowerShell module to be installed and authenticated.
 #>
 
-# Main script
-clear-host
-Write-Output "####################################################################################################"
-Write-Output "## RETRIEVING ALL AVAILABILITIES IN THIS SUBSCRIPTION                                             ##"
-Write-Output "####################################################################################################"
-Write-Output ""
-
-# Initialize REST API
-Write-Output "Initializing REST API"
-Write-Output "  Retrieving access token and subscription ID"
-$AzAccessTokenSecure = Get-AzAccessToken -AsSecureString
-$SecureToken = $AzAccessTokenSecure.Token
-$REST_AccessToken = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($SecureToken))
-$REST_SubscriptionId = (Get-AzContext).Subscription.Id
-Write-Output "    Access token and Subscription ID successfully retrieved"
-Write-Output "    Subscription ID: $REST_SubscriptionId"
-Write-Output "  Setting up BaseUri and headers"
-$REST_BaseUri = "https://management.azure.com/subscriptions"
-$REST_Headers = @{
-    Authorization = "Bearer $REST_AccessToken"
+function Out-JSONFile {
+    param (
+        [Parameter(Mandatory = $true)]
+        [object]$Data,
+        [Parameter(Mandatory = $true)]
+        [string]$FileName
+    )
+    # This function writes the provided data to a JSON file at the specified path.
+    Write-Output "  Writing data to file: $FileName"
+    $Data | ConvertTo-Json -Depth 20 | Out-File -FilePath "$(Get-Location)\$FileName" -Force
 }
 
-# Providers: Start
-Write-Output "Retrieving all available provider"
-$Providers_Uri = "$REST_BaseUri/$REST_SubscriptionId/providers?api-version=2021-04-01" 2>$null
-$Providers_Response = Invoke-RestMethod -Uri $Providers_Uri -Headers $REST_Headers -Method Get
-
-# Transform the response to the desired structure
-$Resources_All = foreach ($provider in $Providers_Response.value) {
-    # Build an array of resource types using plain hashtables
-    $rtArray = @()
-    foreach ($rt in $provider.resourceTypes) {
-        $rtArray += @{
-            Type      = $rt.resourceType
-            Locations = $rt.locations
-        }
+function Write-Headline {
+    param (
+    [Parameter(Mandatory = $true)]
+    [object]$Text
+    )
+    Write-Output "####################################################################################################"
+    Write-Output "   $Text"
+    Write-Output "####################################################################################################"
+    Write-Output ""
+}
+function Initialize-RESTAPI {
```

Comment:
Consider removing this function and use Invoke-AzRestMethod rather than invoke-Restmethod in the overall script